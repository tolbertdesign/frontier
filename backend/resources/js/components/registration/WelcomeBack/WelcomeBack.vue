<template>
  <div>
    <h1 class="text-48 mw-200px mx-auto">
      {{ lang.welcome_back }}
    </h1>
    <p class="mt-12">
      <a
        v-if="isBetaUser && !isOrgAdmin"
        href="/v3/home/dashboard"
        class="btn btn-white-outline btn-813-dk-grey-outline btn-round mb-4 py-2 d-block mw-200px mx-auto text-15 w-200px"
      >{{ lang.back_to_dashboard }}</a>
      <a
        v-if="isOrgAdmin"
        :href="`/v3/tkdashboard/?redirect=admin`"
        class="btn btn-white-outline btn-813-dk-grey-outline btn-round mb-4 py-2 d-block mw-200px mx-auto text-15 w-200px"
      >{{ lang.admin_dashboard }}</a>
      <button
        class="btn btn-primary btn-round py-2 d-block mw-200px mx-auto mb-4 btn-drop-shadow text-15 w-200px"
        @click="_setUserType('parent')"
      >{{ buttonText }}</button>
      <button
        class="btn btn-success btn-round py-2 d-block mw-200px mx-auto mb-4 text-15 w-200px"
        @click="_setUserType('teacher')"
      >{{ lang.im_a_teacher }}</button>
      <a
        data-toggle="modal"
        data-target="#sponsorInstructionModal"
        href="#"
        class="btn btn-white-outline btn-813-dk-grey-outline btn-round py-2 d-block mw-200px mx-auto text-15 w-200px"
        @click="_sponsor"
      >{{ lang.im_a_sponsor }}</a>

      <a
        id="regButtonsExplanationModalAnchor"
        data-toggle="modal"
        data-target="#regButtonsExplanationModal"
        href="#"
        class="py-3 d-block mw-200px mx-auto text-15 text-white font-weight-bold"
        @click="_whatDoesThisMean"
      >{{ lang.what_does_this_mean }}</a>

    </p>
  </div>
</template>

<script>
export default {
  props: {
    isBetaUser: {
      type: Boolean,
      default: false,
    },
    isOrgAdmin: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    lang () {
      return this.$store.state.lang
    },
    buttonText () {
      if (window.location.hostname.includes('funrunpro')) {
        return this.lang.im_a_participant
      }
      return this.lang.im_a_parent
    },
  },
  mounted: function () {
    this.hjTrigger('dash-welcome-back')
  },
  methods: {
    _setUserType (val) {
      this.gaEvent('Titan Registration', 'Welcome Back', val)
      axios.get('/v3/api/validate-profile-complete')
        .then((response) => {
          if (response.data.valid === true) {
            this.$store.commit('SET_USER_TYPE', val)
            this.$emit('selection-made')
          } else {
            this.$store.commit('SET_USER_TYPE', 'incompleteProfile')
            this.$emit('incomplete-registration')
          }
        })
    },
    _sponsor () {
      this.gaEvent('Titan Registration', 'Welcome Back', 'sponsor')
    },
    _whatDoesThisMean () {
      this.gaEvent('Titan Registration', 'Welcome Back', 'What does this mean')
    },
  },
}

</script>
