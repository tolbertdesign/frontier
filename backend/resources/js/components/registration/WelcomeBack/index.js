import WelcomeBack from './WelcomeBack.vue'

export default WelcomeBack
