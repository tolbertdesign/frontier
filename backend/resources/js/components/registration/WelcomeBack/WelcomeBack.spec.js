import { shallowMount, createLocalVue } from '@vue/test-utils'
import WelcomeBack from './WelcomeBack.vue'

const localVue = createLocalVue()

const $store = {
  state: {
    lang: {
      welcome_back: 'welcome back',
      back_to_dashboard: 'back to dashboard',
      admin_dashboard: 'admin dashboard',
      im_a_teacher: 'I am a teacher',
      im_a_parent: 'I am a parent',
      im_a_student: 'I am a student',
      im_a_sponsor: 'I am a sponsor',
      what_does_this_mean: 'what does this mean',
    },
  },
}
const isBetaUser = false
const isOrgAdmin = false

describe('WelcomeBack', () => {
  it('Snapshop test for default WelcomeBack', () => {
    const wrapper = shallowMount(WelcomeBack, {
      localVue,
      mocks: {
        $store: $store,
        hjTrigger: jest.fn(),
        gaEvent: jest.fn(),
      },
      propsData: {
        isBetaUser: isBetaUser,
        isOrgAdmin: isOrgAdmin,
      },
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.vm.buttonText).toBe($store.state.lang.im_a_parent)
  })
  it('Snapshop test for beta user WelcomeBack', () => {
    const wrapper = shallowMount(WelcomeBack, {
      localVue,
      mocks: {
        $store: $store,
        hjTrigger: jest.fn(),
        gaEvent: jest.fn(),
      },
      propsData: {
        isBetaUser: true,
        isOrgAdmin: isOrgAdmin,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
  it('Snapshop test for org admin user WelcomeBack', () => {
    const wrapper = shallowMount(WelcomeBack, {
      localVue,
      mocks: {
        $store: $store,
        hjTrigger: jest.fn(),
        gaEvent: jest.fn(),
      },
      propsData: {
        isBetaUser: isBetaUser,
        isOrgAdmin: true,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
  it('Snapshop test for org admin and beta banner user WelcomeBack', () => {
    const wrapper = shallowMount(WelcomeBack, {
      localVue,
      mocks: {
        $store: $store,
        hjTrigger: jest.fn(),
        gaEvent: jest.fn(),
      },
      propsData: {
        isBetaUser: true,
        isOrgAdmin: true,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
