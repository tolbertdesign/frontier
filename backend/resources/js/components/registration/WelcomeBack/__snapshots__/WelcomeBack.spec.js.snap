// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeBack Snapshop test for beta user WelcomeBack 1`] = `
<div>
  <h1 class="text-48 mw-200px mx-auto">welcome back</h1>
  <p class="mt-12"><a href="/v3/home/dashboard" class="btn btn-white-outline btn-813-dk-grey-outline btn-round mb-4 py-2 d-block mw-200px mx-auto text-15 w-200px">back to dashboard</a>
    <!----> <button class="btn btn-primary btn-round py-2 d-block mw-200px mx-auto mb-4 btn-drop-shadow text-15 w-200px">I am a parent</button> <button class="btn btn-success btn-round py-2 d-block mw-200px mx-auto mb-4 text-15 w-200px">I am a teacher</button> <a data-toggle="modal" data-target="#sponsorInstructionModal" href="#" class="btn btn-white-outline btn-813-dk-grey-outline btn-round py-2 d-block mw-200px mx-auto text-15 w-200px">I am a sponsor</a> <a id="regButtonsExplanationModalAnchor" data-toggle="modal" data-target="#regButtonsExplanationModal" href="#" class="py-3 d-block mw-200px mx-auto text-15 text-white font-weight-bold">what does this mean</a></p>
</div>
`;

exports[`WelcomeBack Snapshop test for default WelcomeBack 1`] = `
<div>
  <h1 class="text-48 mw-200px mx-auto">welcome back</h1>
  <p class="mt-12">
    <!---->
    <!----> <button class="btn btn-primary btn-round py-2 d-block mw-200px mx-auto mb-4 btn-drop-shadow text-15 w-200px">I am a parent</button> <button class="btn btn-success btn-round py-2 d-block mw-200px mx-auto mb-4 text-15 w-200px">I am a teacher</button> <a data-toggle="modal" data-target="#sponsorInstructionModal" href="#" class="btn btn-white-outline btn-813-dk-grey-outline btn-round py-2 d-block mw-200px mx-auto text-15 w-200px">I am a sponsor</a> <a id="regButtonsExplanationModalAnchor" data-toggle="modal" data-target="#regButtonsExplanationModal" href="#" class="py-3 d-block mw-200px mx-auto text-15 text-white font-weight-bold">what does this mean</a></p>
</div>
`;

exports[`WelcomeBack Snapshop test for org admin and beta banner user WelcomeBack 1`] = `
<div>
  <h1 class="text-48 mw-200px mx-auto">welcome back</h1>
  <p class="mt-12">
    <!----> <a href="/v3/tkdashboard/?redirect=admin" class="btn btn-white-outline btn-813-dk-grey-outline btn-round mb-4 py-2 d-block mw-200px mx-auto text-15 w-200px">admin dashboard</a> <button class="btn btn-primary btn-round py-2 d-block mw-200px mx-auto mb-4 btn-drop-shadow text-15 w-200px">I am a parent</button> <button class="btn btn-success btn-round py-2 d-block mw-200px mx-auto mb-4 text-15 w-200px">I am a teacher</button> <a data-toggle="modal" data-target="#sponsorInstructionModal" href="#" class="btn btn-white-outline btn-813-dk-grey-outline btn-round py-2 d-block mw-200px mx-auto text-15 w-200px">I am a sponsor</a> <a id="regButtonsExplanationModalAnchor" data-toggle="modal" data-target="#regButtonsExplanationModal" href="#" class="py-3 d-block mw-200px mx-auto text-15 text-white font-weight-bold">what does this mean</a></p>
</div>
`;

exports[`WelcomeBack Snapshop test for org admin user WelcomeBack 1`] = `
<div>
  <h1 class="text-48 mw-200px mx-auto">welcome back</h1>
  <p class="mt-12">
    <!----> <a href="/v3/tkdashboard/?redirect=admin" class="btn btn-white-outline btn-813-dk-grey-outline btn-round mb-4 py-2 d-block mw-200px mx-auto text-15 w-200px">admin dashboard</a> <button class="btn btn-primary btn-round py-2 d-block mw-200px mx-auto mb-4 btn-drop-shadow text-15 w-200px">I am a parent</button> <button class="btn btn-success btn-round py-2 d-block mw-200px mx-auto mb-4 text-15 w-200px">I am a teacher</button> <a data-toggle="modal" data-target="#sponsorInstructionModal" href="#" class="btn btn-white-outline btn-813-dk-grey-outline btn-round py-2 d-block mw-200px mx-auto text-15 w-200px">I am a sponsor</a> <a id="regButtonsExplanationModalAnchor" data-toggle="modal" data-target="#regButtonsExplanationModal" href="#" class="py-3 d-block mw-200px mx-auto text-15 text-white font-weight-bold">what does this mean</a></p>
</div>
`;
